src/Microsoft.SqlTools.ServiceLayer/Scripting/ScriptAsOptions.cs

```diff
@@ -0,0 +1,712 @@
+        /// </summary>
+        /// <remarks>
+        /// The default is true.
+        /// </remarks>
+        public override bool? SchemaQualify
+        {
+            get
+            {
+                return ScriptOptions.SchemaQualify;
+            }
+            set
+            {
+                ScriptOptions.SchemaQualify = value;
+            }
+        }
+
+        /// <summary>
+        /// Script options to set bindings option.
+        /// </summary>
+        public override bool? Bindings
+        {
+            get
+            {
+                return ScriptOptions.Bindings;
+            }
+            set
+            {
+                ScriptOptions.Bindings = value;
+            }
+        }
+
+        /// <summary>
+        /// Script the objects that use collation.
+        /// </summary>
+        public override bool? Collation
+        {
+            get
+            {
+                return ScriptOptions.Collation;
+            }
+            set
+            {
+                ScriptOptions.Collation = value;
+            }
+        }
+
+        /// <summary>
+        /// Script the default values.
+        /// </summary>
+        /// <remarks>
+        /// The default is true.
+        /// </remarks>
+        public override bool? Default
+        {
+            get
+            {
+                return ScriptOptions.Default;
+            }
+            set
+            {
+                ScriptOptions.Default = value;
+            }
+        }
+
+        /// <summary>
+        /// Script Object CREATE/DROP statements.  
+        /// Possible values: 
+        ///   ScriptCreate
+        ///   ScriptDrop
+        ///   ScriptCreateDrop
+        ///   ScriptSelect
+        /// </summary>
+        /// <remarks>
+        /// The default is ScriptCreate.
+        /// </remarks>
+        public override string ScriptCreateDrop
+        {
+            get
+            {
+                return ScriptOptions.ScriptCreateDrop;
+            }
+            set
+            {
+                ScriptOptions.ScriptCreateDrop = value;
+            }
+        }
+
+        /// <summary>
+        /// Script the Extended Properties for each object scripted.
+        /// </summary>
+        /// <remarks>
+        /// The default is true.
+        /// </remarks>
+        public override bool? ScriptExtendedProperties
+        {
+            get
+            {
+                return ScriptOptions.ScriptExtendedProperties;
+            }
+            set
+            {
+                ScriptOptions.ScriptExtendedProperties = value;
+            }
+        }
+
+        /// <summary>
+        /// Script only features compatible with the specified version of SQL Server.  Possible values:
+        ///   Script90Compat
+        ///   Script100Compat
+        ///   Script105Compat
+        ///   Script110Compat
+        ///   Script120Compat
+        ///   Script130Compat
+        ///   Script140Compat  
```

Comment:
Script150Compat should be present too... right? Or is that not available yet...?

Reply:
not sure if it's available, I don't see it as a valid option in SM. I'll verify ylater

---
In reply to: [154440666](https://github.com/Microsoft/sqltoolsservice/pull/556#discussion_r154440666) [](ancestors = 154440666)